SofaKernel/modules/SofaBaseMechanics/src/SofaBaseMechanics/DiagonalMass.h

```diff
@@ -244,6 +244,9 @@ class DiagonalMass : public core::behavior::Mass<DataTypes>
     void initTopologyHandlers();
     void massInitialization();
 
+    /// Will compute the vertexMass using input denstiy and return the corresponding full mass.
+    Real computeVertexMass(const Real& density);
```

Comment:
```suggestion
    Real computeVertexMass(Real density);
```
Pass primitive types by value